### PR TITLE
Update ogrmerge.rst making ready for primetime

### PR DESCRIPTION
**Pull request by jidanni**
I tested it and it works great.
```
$ ogrmerge --version
GDAL 3.9.0, released 2024/05/07
$ man ogrmerge | wc
    153     765    7272
$ man ogrmerge.py | wc
No manual entry for ogrmerge.py
      0       0       0
```

### CODE REVIEW

**Comment by jratike80** [comment]
What certainly works for running Python scripts is syntax `python path_to_script`.

```
python  C:\OSGeo4W\apps\Python312\Scripts\ogrmerge.py
Usage: ogrmerge.py [--help] [--help-general]
...
```

There may be helper systems which make is possible to launch the script with easier syntax  `ogrmerge` but that is not guaranteed to work everywhere.

**Comment by jidanni** [comment]
All I know is one doesn't want to hardwire "fortran", etc, into filenames that might get rewritten in a different language in the future.

**Comment by jratike80** [comment]
But they are Python scripts and not compiled programs like gdalinfo.  Users must have a compatible Python version on their own computer for using those utilities. Internally Python must be started first before the script can be run. It is important to know that distinction.

If the script is rewritten in C++ some day the .py will naturally disappear from the filename.

**Comment by jidanni** [comment]
Certainly such users not playing with a full deck will learn very quickly via error messages telling them to mend their ways.

Else with ".sh", ".this", ".that" extensions hanging out all over the place, GDAL/OGR will look like an unfinished college project. 

And as a matter of fact, of the thousands of programs in my /usr/bin , guess which are the only ones that even have any 'language extension' at all. It makes me wonder if they somehow slipped in against some Debian policy or rules of style or something.

**Comment by jidanni** [comment]
A N D    T H E R E   you have it:

https://www.debian.org/doc/debian-policy/ch-files.html

> When scripts are installed into a directory in the system PATH, the script name should not include an extension such as .sh or .pl that denotes the scripting language currently used to implement it.

**Comment by jratike80** [comment]
After this https://github.com/OSGeo/gdal/pull/5296 a part of the Python scripts can be launched without .py extension. See also https://github.com/OSGeo/gdal/issues/5281

The change does not affect older GDAL versions but I guess that the simple syntax that you suggest is OK but I would like to have some Python expert to confirm. 

> Certainly such users not playing with a full deck will learn very quickly via error messages telling them to mend their ways.

Glad that you also have trust in our users. I would still keep the .py for example on this list so the users can quickly see which tools require Python and GDAL Python bindings, and which not.

image

**Comment by jidanni** [comment]
> I would still keep the .py for example on this list so the users can quickly see which tools require Python and GDAL Python bindings, and which not.

OK, but change
> bla_bla.py

to
> bla_bla (py)

or
> bla_bla (python)

please.

**Comment by jratike80** [comment]
Maybe there should be then a clear mention about tool being a Python script.

Now
> gdal_calc.py
> Command line raster calculator with numpy syntax.

Then 
> gdal_calc
> Command line raster calculator with numpy syntax. The tool is a Python script that requires GDAL Python bindings https://gdal.org/api/python_bindings.html

File: doc/source/programs/ogrmerge.rst [hunk 1]
```diff
@@ -183,18 +183,24 @@ Create a VRT with a layer for each input shapefiles
...
 
 .. code-block::
 
-    ogrmerge.py -single -o merged.shp france.shp germany.shp -src_layer_field_name country
+    ogrmerge -single -o merged.shp france.shp germany.shp -src_layer_field_name country
+
+Notes
+-----
+
+Older versions of ogrmerge were called ogrmerge.py.
```
Comment (rouault) [inline doc/source/programs/ogrmerge.rst:205]:
```suggestion
ogrmerge is a Python utility, and is only available if GDAL Python bindings are available.
```

**Pull request merged**